lib/web_ui/lib/src/engine/surface/backdrop_filter.dart

```diff
@@ -65,14 +65,41 @@ class PersistedBackdropFilter extends PersistedContainerSurface
       _invertedTransform = Matrix4.inverted(_transform);
       _previousTransform = _transform;
     }
-    final ui.Rect rect = transformLTRB(_invertedTransform, 0, 0,
-        ui.window.physicalSize.width, ui.window.physicalSize.height);
+    // https://api.flutter.dev/flutter/widgets/BackdropFilter-class.html
+    // Defines the effective area as the parent/ancestor clip or if not
+    // available, the whole screen.
+    //
+    // The CSS backdrop-filter will use isolation boundary defined in
+    // https://drafts.fxtf.org/filter-effects-2/#BackdropFilterProperty
+    // Therefore we need to use parent clip element bounds for
+    // backdrop boundary.
+    final double dpr = ui.window.devicePixelRatio;
+    ui.Rect rect = transformLTRB(_invertedTransform, 0, 0,
+        ui.window.physicalSize.width * dpr, ui.window.physicalSize.height);
+    PersistedSurface parentSurface = parent;
+    String left = '{rect.left}px';
+    String top = '{rect.top}px';
+    String width = '${rect.width}px';
+    String height = '${rect.height}px';
+    while (parentSurface != null) {
+      html.HtmlElement parentElement = parentSurface.rootElement;
```

Comment:
Actually, I think I know the proper place for the logic that figures out the nearest clip: `recomputeTransformAndClip`. It is invoked as part of the `preroll` to figure out clip changes even for `addRetained`. This way, the backdrop filter doesn't need to walk up the parent chain at all. Instead, it will be given nearest clip, so it can react appropriately when the clip changes.

Reply:
Done. Reading localClipRect from PersistedContainerSurface instead.